packages/core/src/context/layer1.ts

```diff
@@ -216,13 +220,31 @@ export class L1Contract extends ZkopruContract {
     account: Account,
     option?: Tx,
   ): Promise<TransactionReceipt | undefined> {
+    const done = await this.aquireSendTxLock(account.address)
```

Comment:
The `async-lock` package allows you to return a promise. This function can be refactored like this:

```
const result = await this.sendTxLock.acquire(account.address, () => TxUtil.sendTx(
  tx,
  this.address,
  this.web3,
  account,
  options
))
return result
```

Reply:
Awesome! much better, thanks.
I was looking for something simpler like this one but did not get it.